pkg/build/strategies/layered/layered.go

```diff
@@ -41,10 +42,13 @@ type Layered struct {
 }
 
 // New creates a Layered builder.
-func New(config *api.Config, scripts build.ScriptsHandler, overrides build.Overrides) (*Layered, error) {
+func New(config *api.Config, scripts build.ScriptsHandler, overrides build.Overrides) (*Layered, []*api.BuildStepInfo, error) {
```

Comment:
I find it strange that, in s2i itself, all calls to `New` and `Strategy` and `GetStrategy` ignore the second return value...

IIUC, all that `[]*api.BuildStepInfo` conveys in a list of errors.
If that's the case, wouldn't it be a cleaner design to keep the function signature as it was, and make `[]*api.BuildStepInfo` implement the `error` interface?

If what I'm saying makes sense, sorry if it doesn't, we could store the original error within the "FailureReason", thus this PR would be about wrapping errors allowing for easy identification of what step went wrong and why.

WDYT?